tools/src/main/scala/scala/scalanative/build/NativeConfig.scala

```diff
@@ -0,0 +1,156 @@
+package scala.scalanative
+package build
+
+import java.nio.file.{Path, Paths}
+
+/** An object describing how to configure the Scala Native toolchain. */
+sealed trait NativeConfig {
+
+  /** The garbage collector to use. */
+  def gc: GC
+
+  /** Compilation mode. */
+  def mode: Mode
+
+  /** The path to the `clang` executable. */
+  def clang: Path
+
+  /** The path to the `clang++` executable. */
+  def clangPP: Path
+
+  /** The options passed to LLVM's linker. */
+  def linkingOptions: Seq[String]
+
+  /** The compilation options passed to LLVM. */
+  def compileOptions: Seq[String]
+
+  /** Should stubs be linked? */
+  def linkStubs: Boolean
+
+  /** The LTO mode to use used during a release build. */
+  def LTO: LTO
```

Comment:
Prefer lowercase acronyms when they are at the beginning of a `camelCase` identifier.
```suggestion
  def lto: LTO
```
This is similar to `def gc: GC` for example.